support/util/scheduler.go

```diff
@@ -0,0 +1,148 @@
+		hc.Annotations[hyperv1.APICriticalPriorityClass] = *sizeConfig.Effects.APICriticalPriorityClassName
+	}
+	if sizeConfig.Effects != nil && sizeConfig.Effects.MachineHealthCheckTimeout != nil {
+		hc.Annotations[hyperv1.MachineHealthCheckTimeoutAnnotation] = sizeConfig.Effects.MachineHealthCheckTimeout.Duration.String()
+	} else {
+		// If mhc timeout is configured for any size in the config, remove the annotation
+		// to fallback to the default
+		if configHasMHCTimeout(config) {
+			delete(hc.Annotations, hyperv1.MachineHealthCheckTimeoutAnnotation)
+		}
+	}
+	if sizeConfig.Effects != nil && sizeConfig.Effects.MaximumRequestsInflight != nil {
+		hc.Annotations[hyperv1.KubeAPIServerMaximumRequestsInFlight] = fmt.Sprint(*sizeConfig.Effects.MaximumRequestsInflight)
+	}
+	if sizeConfig.Effects != nil && sizeConfig.Effects.MaximumMutatingRequestsInflight != nil {
+		hc.Annotations[hyperv1.KubeAPIServerMaximumMutatingRequestsInFlight] = fmt.Sprint(*sizeConfig.Effects.MaximumMutatingRequestsInflight)
+	}
+
+	var resourceRequestAnnotations map[string]string
+	if sizeConfig.Effects != nil {
+		resourceRequestAnnotations = ResourceRequestsToOverrideAnnotations(sizeConfig.Effects.ResourceRequests)
+	}
+	for k, v := range resourceRequestAnnotations {
+		hc.Annotations[k] = v
+	}
+
+	//For ROSA, get the subnets from the nodes and set the annotation
+	if nodes != nil {
+		lbSubnets := getLBSubnetsFromNodes(nodes)
+		if lbSubnets != "" {
+			hc.Annotations[hyperv1.AWSLoadBalancerSubnetsAnnotation] = lbSubnets
+		}
+
+		hc.Annotations[hyperv1.RequestServingNodeAdditionalSelectorAnnotation] = fmt.Sprintf("%s=%s", hyperv1.NodeSizeLabel, size)
+
+	}
+
+	return hc, nil
+}
+
+func getLBSubnetsFromNodes(nodes []corev1.Node) string {
+	lbSubnets := ""
+	for _, node := range nodes {
+		if node.Labels[LBSubnetsLabel] != "" {
+			lbSubnets = node.Labels[LBSubnetsLabel]
+			break
+		}
+	}
+	if lbSubnets != "" {
+		// If subnets are separated by periods, replace them with commas
+		lbSubnets = strings.ReplaceAll(lbSubnets, ".", ",")
+		return lbSubnets
+	}
+	return ""
+}
+
+func getGoMemLimitLabelFromNodes(nodes []corev1.Node) string {
+	for _, node := range nodes {
+		if node.Labels[GoMemLimitLabel] != "" {
+			return node.Labels[GoMemLimitLabel]
+		}
+	}
+	return ""
+}
+
+func SizeConfiguration(config *schedulingv1alpha1.ClusterSizingConfiguration, size string) *schedulingv1alpha1.SizeConfiguration {
+	for i := range config.Spec.Sizes {
+		if config.Spec.Sizes[i].Name == size {
+			return &config.Spec.Sizes[i]
+		}
+	}
+	return nil
+}
+
+func configHasMHCTimeout(config *schedulingv1alpha1.ClusterSizingConfiguration) bool {
+	for _, size := range config.Spec.Sizes {
+		if size.Effects != nil && size.Effects.MachineHealthCheckTimeout != nil {
+			return true
+		}
+	}
+	return false
+}
+
+func ResourceRequestsToOverrideAnnotations(requests []schedulingv1alpha1.ResourceRequest) map[string]string {
```

Comment:
doesn't look like this needs to be public

Reply:
It needed here https://github.com/openshift/hypershift/pull/5417/files#diff-be4da6a931f52746ccd6902dca5fd9936ac0ffb1179417b9577d970aa3ce7bb4R756